python/ray/data/_internal/stats.py

```diff
@@ -494,24 +492,28 @@ def from_block_metadata(
 
         Args:
             block_metas: List of `BlockMetadata` to calculate stats of
-            time_total_s: Total execution time of stage
             stage_name: Name of stage associated with `blocks`
             is_substage: Whether this set of blocks belongs to a substage.
         Returns:
             A `StageStatsSummary` object initialized with the calculated statistics
         """
         exec_stats = [m.exec_stats for m in block_metas if m.exec_stats is not None]
+        rounded_total = 0
+        time_total_s = 0
 
         if is_substage:
             exec_summary_str = "{}/{} blocks executed\n".format(
                 len(exec_stats), len(block_metas)
             )
         else:
-            rounded_total = round(time_total_s, 2)
-            if rounded_total <= 0:
-                # Handle -0.0 case.
-                rounded_total = 0
             if exec_stats:
+                # Calculate the total execution time of stage as
+                # the maximum wall time from all blocks' stats.
```

Comment:
Would this give us the right stage time if there were multiple rounds of blocks?

I was thinking this would be something like max_block_finish_time - min_block_start_time?

Reply:
> Would this give us the right stage time if there were multiple rounds of blocks?
@stephanie-wang would this be for the case where there are multiple substages? or what does "multiple rounds of blocks" mean in this case?

If the former, then I'm thinking we can include additional logic in `StageStatsSummary.from_block_metadata()` (or a separate function) that can handle summing times across substages. 

Also, what attribute(s) can I reference for calculating `min/max_block_start_time`?

Reply:
Ah, I thought this number was the max duration of a single block/task. If all tasks in the stage run in parallel, then that will be the same as the stage time, but if we have to run multiple parallel rounds of tasks, then the actual stage time will be longer.

> Also, what attribute(s) can I reference for calculating min/max_block_start_time?

Hmm it looks like we don't track this right now. I guess ideally each block would report its start and end time, in addition to max duration.

Reply:
Yea, we should probably track the start/end time. It's a bit sketchy since clocks might not be quite in sync, but it is useful for doing rough calculations like in this case.